pkg/sql/internal_test.go

```diff
@@ -714,6 +714,23 @@ func TestInternalExecutorRetryAfterRows(t *testing.T) {
 	if !testutils.IsError(err, "inject_retry_errors_enabled") {
 		t.Fatalf("expected to see injected retry error, got %v", err)
 	}
+
+	// Now verify that ExecEx correctly and transparently to us retries the
+	// stmt.
+	numRows, err := ie.ExecEx(
+		ctx, "read rows", nil, /* txn */
```

Comment:
Should we have a test where txn isn't nil and has already been used to do something and we don't retry in that case?